src/server/frontend/render.js

```diff
@@ -99,7 +104,17 @@ function getAppHtml(store, renderProps) {
   );
 }
 
-function getScriptHtml(clientState, headers, hostname) {
+let appAssetFilenameCache = null;
+
+async function getAppAssetFilenamesCachedAsync() { // eslint-disable-line space-before-function-paren
+  if (appAssetFilenameCache) return appAssetFilenameCache;
+
+  appAssetFilenameCache = await getAppAssetFilenamesAsync();
```

Comment:
What's the reasoning behind naming the stuff `async` ? I mean, sounds like a good pattern, but we do not write `getAppAssetFileNamesPromise()`